packages/web-components/fast-components/src/menu/menu.ts

```diff
@@ -0,0 +1,183 @@
+import { DOM, FASTElement, observable } from "@microsoft/fast-element";
+import { inRange, invert } from "lodash-es";
+import {
+    isHTMLElement,
+    keyCodeArrowDown,
+    keyCodeArrowLeft,
+    keyCodeArrowRight,
+    keyCodeArrowUp,
+    keyCodeEnd,
+    keyCodeHome,
+} from "@microsoft/fast-web-utilities";
+import { MenuItemRole } from "../menu-item";
+
+export class Menu extends FASTElement {
+    @observable
+    public items: HTMLSlotElement;
+    private itemsChanged(): void {
+        this.menuItems = this.domChildren();
+    }
+
+    private menuItems: Element[];
+
+    /**
+     * The index of the focusable element in the items array
+     * defaults to -1
+     */
+    private focusIndex: number = -1;
+
+    private static focusableElementRoles: { [key: string]: string } = invert(
+        MenuItemRole
+    );
+
+    public connectedCallback(): void {
+        super.connectedCallback();
+
+        // We need to queue update currently to ensure that
+        // the attributes on our menu items have time to be set
+        DOM.queueUpdate(() => {
+            const focusIndex = this.menuItems.findIndex(this.isFocusableElement);
+
+            // if our focus index is not -1 we have items
+            if (focusIndex !== -1) {
+                this.focusIndex = focusIndex;
+            }
+
+            for (let item: number = 0; item < this.menuItems.length; item++) {
+                if (item === focusIndex) {
+                    this.menuItems[item].setAttribute("tabindex", "0");
+                }
+
+                this.menuItems[item].addEventListener("blur", this.handleMenuItemFocus);
+            }
+        });
+    }
+
+    public focus(): void {
+        this.setFocus(0, 1);
+    }
+
+    public handleMenuKeyDown(e: KeyboardEvent): void | boolean {
+        switch (e.keyCode) {
+            case keyCodeArrowDown:
+            case keyCodeArrowRight:
+                // if I'm a submenu, open me
+                // if I'm not a submenu, go forward one index
```

Comment:
Since we won't be handling submenus we can likely pull the comments relating to it.